Kernel/Arch/aarch64/Processor.h

```diff
@@ -205,8 +205,7 @@ ALWAYS_INLINE void ProcessorBase<T>::wait_check()
 template<typename T>
 ALWAYS_INLINE u64 ProcessorBase<T>::read_cpu_counter()
 {
-    TODO_AARCH64();
-    return 0;
+    return Aarch64::PMCCNTR_EL0::read().CCNT;
```

Comment:
This register is sadly not required on AArch64. We should check `has_feature(CPUFeature::PMUv3)` before accessing it (https://developer.arm.com/documentation/ddi0601/2021-12/AArch64-Registers/PMCCNTR-EL0--Performance-Monitors-Cycle-Count-Register?lang=en).
So `Processor::read_cpu_counter` should probably return an `Optional<u64>` instead.
You don't need to do this in this PR either but I would prefer if we didn't assume that PMUv3 is present.